src/relations/postgresql_provider.py

```diff
@@ -93,6 +88,19 @@ def _on_database_requested(self, event: DatabaseRequestedEvent) -> None:
             event.defer()
             return
 
+        self.charm.update_config()
```

Comment:
This will update the hash on the leader and trigger relation change event on any follower units.